### PR TITLE
Add Quirk to block filename in item uri

### PR DESCRIPTION
**Pull request by KarlStraussberger**
second part of solution for #2531
may even help with #2017

### CODE REVIEW

File: src/upnp_xml.cc [hunk 1]
```diff
@@ -471,31 +471,31 @@ std::pair<std::string, bool> UpnpXMLBuilder::renderItemImage(const std::string&
     return {};
 }
 
-std::pair<std::string, bool> UpnpXMLBuilder::renderSubtitle(const std::string& virtualURL, const std::shared_ptr<CdsItem>& item)
+std::pair<std::string, bool> UpnpXMLBuilder::renderSubtitle(const std::string& virtualURL, const std::shared_ptr<CdsItem>& item, const std::unique_ptr<Quirks>& quirks)
```
Comment (whyman) [inline src/upnp_xml.cc:474]: Maybe just pass a reference to the Quirks here, not sure it makes sense to pass a reference to a unique_ptr

Comment (KarlStraussberger) [reply src/upnp_xml.cc:474]: Sometimes you do not see the obvious

Comment (KarlStraussberger) [reply src/upnp_xml.cc]: Needed to roll it back, because quirks are nullable. Crashed in tests

Comment (KarlStraussberger) [reply src/upnp_xml.cc]: Pointers work

**Pull request merged**